### PR TITLE
Show default outputs

### PR DESCRIPTION
**Pull request by ncfavier**
Following https://github.com/NixOS/nix/pull/6242, we now have access to default outputs so we can single them out in the frontend.

We can also drop `--out-path`, which speeds up import quite a lot and doesn't require increasing swap space or allowing broken/unfree packages anymore.

Blocked until the next nix release (unless we're impatient and use nix from the latest master)

### CODE REVIEW

**Comment by ncfavier** [comment]
I decided to be impatient and install `nixUnstable` (which now contains https://github.com/NixOS/nix/pull/6242) for the nixpkgs import.

Note that we can now relax the minimum Nix version to 2.4 since we're not using `--out-path`, and indexing outputs is optional.

The 21.11 channel doesn't have outputs for now because I messed up the first import and I don't want to bump VERSION again, so maybe wait until the next nixos-21.11 update.

### Screenshots

screenshot-2022-03-25-112035
screenshot-2022-03-25-112411
screenshot-2022-03-25-112054

File: .github/workflows/import-to-elasticsearch.yml [hunk 1]
```diff
@@ -58,6 +53,11 @@ jobs:
       with:
         CACHIX_SIGNING_KEY: ${{ secrets.CACHIX_SIGNING_KEY }}
 
+    - name: Installing Nix unstable
+      run: |
+        nix-env -f '<nixpkgs>' -iA nixVersions.unstable # for https://github.com/NixOS/nix/pull/6242
+        echo "$HOME/.nix-profile/bin" >> "$GITHUB_PATH"
+
```
Comment (ysndr) [inline .github/workflows/import-to-elasticsearch.yml:60]: Why not add this to the `setup` action?
You can provision specific nix versions with `cachix/setupNix`

File: flake-info/src/commands/nixpkgs_info.rs [hunk 2]
```diff
@@ -22,18 +22,9 @@ pub fn get_nixpkgs_info<T: AsRef<str> + Display>(nixpkgs_channel: T) -> Result<V
...
-        "--out-path",
         "--json",
     ]);
 
-    // Nix might fail to evaluate some disallowed packages
-    let mut env = HashMap::new();
-    env.insert("NIXPKGS_ALLOW_BROKEN".into(), "1".into());
-    env.insert("NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM".into(), "1".into());
-    env.insert("NIXPKGS_ALLOW_UNFREE".into(), "1".into());
-    env.insert("NIXPKGS_ALLOW_INSECURE".into(), "1".into());
```
Comment (ysndr) [inline flake-info/src/commands/nixpkgs_info.rs:34]: Is this a result of the changes to nix?

File: flake-info/src/data/import.rs [hunk 3]
```diff
@@ -125,7 +126,9 @@ pub struct Package {
     pub pname: String,
     pub version: String,
     #[serde(default)]
-    pub outputs: HashMap<String, String>,
+    pub outputs: HashMap<String, Option<String>>,
```
Comment (ysndr) [inline flake-info/src/data/import.rs:129]: Why is it `Option<String>` as value type?

**Review by ysndr — APPROVED** [review]
I understand most of the changes are in `nixos/Nix`

The changes here look fine to me, i just had a few small questions

File: .github/workflows/import-to-elasticsearch.yml (hunk 1 above)
Comment (ncfavier) [reply .github/workflows/import-to-elasticsearch.yml:60]: It's really only needed for the nixpkgs import so there's no need to fetch nixUnstable in every other job.

I know you can specify `install_url` when installing Nix but it seems clearer this way

File: flake-info/src/commands/nixpkgs_info.rs (hunk 2 above)
Comment (ncfavier) [reply flake-info/src/commands/nixpkgs_info.rs:34]: Yes, we're not evaluating packages anymore.

File: flake-info/src/data/import.rs (hunk 3 above)
Comment (ncfavier) [reply flake-info/src/data/import.rs:129]: See https://github.com/NixOS/nix/pull/6242#issuecomment-1066567786, we settled on making the output paths nullable.

File: .github/workflows/import-to-elasticsearch.yml (hunk 1 above)
Comment (ysndr) [reply .github/workflows/import-to-elasticsearch.yml:60]: Ok fair enough

**Comment by ncfavier** [comment]
Started an import, will merge after it succeeds

**Pull request merged**